tensorflow/core/kernels/parse_tensor_op.cc

```diff
@@ -56,6 +58,9 @@ class ParseTensorOp : public OpKernel {
         errors::InvalidArgument("Type mismatch between parsed tensor (",
                                 DataTypeString(output.dtype()), ") and dtype (",
                                 DataTypeString(out_type_), ")"));
+    
+    if (!port::kLittleEndian)
+      auto byteswap_status = ByteSwapTensor(&output);
```

Comment:
You need to do something with this status.

Reply:
Thanks @cantonios !

I tried to wrap the returned status with `OP_REQUIRES_OK()`, but when I ran the regression test suite, I found that certain tensors with data type which currently does not support byte swapping (such as `DT_VARIANT`, see https://github.com/tensorflow/tensorflow/blob/6d6f8b643ec13b72b4a19c053bf8f6fa6f8609a0/tensorflow/core/util/tensor_bundle/byte_swap_tensor.cc#L94-L104) will break this macro in the following two test cases:

```bash
//tensorflow/python/data/kernel_tests:checkpoint_test
//tensorflow/python/kernel_tests:composite_tensor_ops_test
```

I'll investigate more on this issue and get back to you later. Thanks again!

Reply:
Hi @cantonios ,

I added a helper function `IsByteSwappable()` in `byte_swap_tensor.cc` to address the above issue. Could you please review this PR again when you have a chance?

Thank you so much!